src/silverlabnwb/nwb_file.py

```diff
@@ -713,9 +716,23 @@ def read_functional_data(self, folder_path):
                 # ts.set_custom_dataset('channel', channel)
                 # # Save the time offset(s) for this ROI, as a link
                 # ts.set_dataset('pixel_time_offsets', 'link:' + roi['pixel_time_offsets'].name)
-                self.add_time_series_data(ts_name, data=data, times=times,
-                                          kind=TwoPhotonSeries,
-                                          ts_attrs=ts_attrs, data_attrs=data_attrs)
+                if self.labview_version is LabViewVersions.pre2018:
+                    self.add_time_series_data(ts_name, data=data, times=times,
+                                              kind=TwoPhotonSeries,
+                                              ts_attrs=ts_attrs, data_attrs=data_attrs)
+                elif self.labview_version is LabViewVersions.v231:
+                    pixel_time_offsets = PixelTimeOffsets(self.raw_pixel_time_offsets[roi_ind])
+                    all_attrs = dict(ts_attrs)
+                    all_attrs.update(data_attrs)
+                    roi_with_offsets = ROISeriesWithPixelTimeOffsets(name=ts_name,
+                                                                     data=data,
+                                                                     pixel_time_offsets=pixel_time_offsets,
+                                                                     rate=1.0/self.cycle_time,
+                                                                     **all_attrs)
+                    self.nwb_file.add_acquisition(roi_with_offsets)
```

Comment:
This can possibly use `self._add_time_series_data(kind=ROISeriesWithPixelTimeOffsets, ...)`?